bukkit/src/main/java/io/ibj/mcauthenticator/engine/events/ChatEvent.java

```diff
@@ -28,7 +28,7 @@ public void onPlayerChat(AsyncPlayerChatEvent event) {
             Iterator<Player> recipients = event.getRecipients().iterator();
             while(recipients.hasNext()) {
                 User u = instance.getCache().get(recipients.next().getUniqueId());
-                if(!u.authenticated()) recipients.remove();
+                if(u != null && !u.authenticated()) recipients.remove();
```

Comment:
I would want to deny messages going to offline players straight out, so u == null || !u.authenticated() would probably be a better option